ui/de-lib/src/main/java/org/iplantc/de/admin/desktop/client/ontologies/presenter/OntologiesPresenterImpl.java

```diff
@@ -0,0 +1,427 @@
+            @Override
+            public void onSuccess(List<Avu> result) {
+                announcer.schedule(new SuccessAnnouncementConfig(appearance.appAvusCleared(targetApp)));
+            }
+        });
+    }
+
+    @Override
+    public void onViewOntologyVersion(ViewOntologyVersionEvent event) {
+        getOntologies();
+    }
+
+    void getOntologies() {
+        serviceFacade.getOntologies(new AsyncCallback<List<Ontology>>() {
+            @Override
+            public void onFailure(Throwable caught) {
+                ErrorHandler.post(caught);
+            }
+
+            @Override
+            public void onSuccess(List<Ontology> result) {
+                Collections.reverse(result);
+                view.showOntologyVersions(result);
+            }
+        });
+    }
+
+    @Override
+    public void onSelectOntologyVersion(SelectOntologyVersionEvent event) {
+        view.clearStore();
+        iriToHierarchyMap.clear();
+        serviceFacade.getOntologyHierarchies(event.getSelectedOntology().getVersion(), new AsyncCallback<List<OntologyHierarchy>>() {
+            @Override
+            public void onFailure(Throwable caught) {
+                ErrorHandler.post(caught);
+            }
+
+            @Override
+            public void onSuccess(List<OntologyHierarchy> result) {
+                if (result.size() == 0) {
+                    view.showEmptyTreePanel();
+                } else {
+                    addHierarchies(null, result);
+                    view.showTreePanel();
+                }
+            }
+        });
+
+    }
+
+    void addHierarchies(OntologyHierarchy parent, List<OntologyHierarchy> children) {
+        if ((children == null)
+            || children.isEmpty()) {
+            return;
+        }
+        if (parent == null) {
+            ontologyUtil.addUnclassifiedChild(children);
+            view.addToStore(children);
+
+        } else {
+            view.addToStore(parent, children);
+        }
+
+        helperMap(children);
+
+        for (OntologyHierarchy hierarchy : children) {
+            addHierarchies(hierarchy, hierarchy.getSubclasses());
+        }
+    }
+
+    void helperMap(List<OntologyHierarchy> children) {
+        for (OntologyHierarchy hierarchy : children) {
+            String iri = hierarchy.getIri();
+            List<OntologyHierarchy> hierarchies = iriToHierarchyMap.get(iri);
+            if (hierarchies == null) {
+                hierarchies = Lists.newArrayList();
+            }
+            hierarchies.add(hierarchy);
+            iriToHierarchyMap.put(hierarchy.getIri(), hierarchies);
+        }
+    }
+
+    @Override
+    public void onSaveOntologyHierarchy(SaveOntologyHierarchyEvent event) {
+        List<String> iris = event.getIris();
+
+        for (final String iri : iris) {
+            serviceFacade.saveOntologyHierarchy(event.getOntology().getVersion(),
+                                                iri, new SaveHierarchyAsyncCallback(iri));
```

Comment:
The tree panel should probably have a "Loading" mask while the service calls are processed.
Otherwise, if the service call takes a while, the admin might not know if the request is still processing or not, or failed without an error message.